### DISCUSSION TITLE
Blast settings for ITS analysis

### DISCUSSION THREAD

**Discussion by marchoeppner** [bio-raum/FooDMe2#10] [General]
@gregdenay 

Been looking at some ITS data and realized that the default blast settings won't work here - depending a little bit on the exact location of the amplicon.

For example: https://onlinelibrary.wiley.com/cms/asset/efaca041-c0eb-490a-9c83-da92799fea9f/men13387-fig-0001-m.jpg

Basically, the primer locations are outside of the ITS region, which means of the ~300bp amplicon, only maybe 180-200bp are actually inside the ITS2 region. 

For Blast, we are setting:

``` BASH
blast_qcov              = "100"
``` 

The UNITE database only contains the actual ITS sequences, so that particular filter will always fail since the query coverage will be << 100%. 

Any ideas? Not quite sure where to address this... We could set blast settings per primer set in
https://github.com/bio-raum/FooDMe2/blob/dev/conf/primers.config

but this will be fairly intransparent, especially if users want to use their own primers. 

This is indeed a situation where using all of GenBank or Refseq over streamlined databases would be an advantage... but maybe, since we use very focused databases, the query coverage argument isn't needed...?

But it is actually even worse for Pacbio, since the amplicon covers multiple ITS regions.

**Comment by gregdenay** [comment]
You mean that only part of the amplicon is covered by the database?
I think this should be absolutely clear for the users. For me it would be a reason to use another database. 
If users chose a sequence outside ITS2, we have to assume there is a good reason for it

**Reply by marchoeppner** [reply]
Correct, the amplicon is larger than the target sequence in UNITE.

**Reply by gregdenay** [reply]
I say no on reducing default qcov. It should be a conscious choice from the user

**Reply by marchoeppner** [reply]
right, well in that case we cannot use UNITE but need to switch to Genbank or RefSeq Mitochondria. I'll add Refseq to do some testing.

**Reply by marchoeppner** [reply]
Nevermind that, ITS is not included in the mitochondrial database (duh). I guess it'll have to be GenBank nt then ... 

I also added NCBI ITS eukaryotes from https://ftp.ncbi.nlm.nih.gov/blast/db

The latest commit now includes a new command line option

```NEXTFLOW
nextflow run bio-raum/FooDMe2 -r dev --list_dbs
```

To list the growing number of supported databases. 

Finally, the installation of GenBank nt can be skipped using:

```NEXTFLOW
nextflow run bio-raum/FooDMe2 -profile my_profile --build_references --run_name build --skip_genbank
```
